e2e/cypress/integration/channel/channel_routing_spec.js

```diff
@@ -0,0 +1,76 @@
+// Copyright (c) 2015-present Mattermost, Inc. All Rights Reserved.
+// See LICENSE.txt for license information.
+
+// ***************************************************************
+// - [#] indicates a test step (e.g. # Go to a page)
+// - [*] indicates an assertion (e.g. * Check the title)
+// - Use element ID when selecting an element. Create one if none.
+// ***************************************************************
+
+describe('Channel routing', () => {
+    before(() => {
+        cy.apiLogin('user-1');
+        cy.visit('/');
+    });
+
+    it('should go to town square channel view', () => {
+        cy.visit('/ad-1/channels/town-square');
+        cy.get('#channelHeaderTitle').should('be.visible').should('contain', 'Town Square');
+    });
+
+    it('should go to private channel view', () => {
+        cy.getCurrentTeamId().then((teamId) => {
+            cy.apiCreateChannel(teamId, 'private-channel', 'Private channel', 'P').then((response) => {
+                cy.visit(`/ad-1/channels/${response.body.name}`);
+                cy.get('#channelHeaderTitle').should('be.visible').should('contain', 'Private channel');
+                cy.apiDeleteChannel(response.body.id);
+            });
```

Comment:
Use convention comments from lines 5-8, something like:

```js
            // # create a private channel
            cy.apiCreateChannel(teamId, 'private-channel', 'Private channel', 'P').then((response) => {
                 // * ensure private channel exists
                cy.visit(`/ad-1/channels/${response.body.name}`);
                cy.get('#channelHeaderTitle').should('be.visible').should('contain', 'Private channel');
                // # remove created private channel
                cy.apiDeleteChannel(response.body.id);
            });
```

Reply:
also, wondering if you should create a random name for the channel, that way if something goes wrong while testing (this or another test is failing) and doesn't clean up the channel, it will fail constantly due to having the name taken

Reply:
All the channel names are random because of [this](https://github.com/mattermost/mattermost-webapp/blob/master/e2e/cypress/support/api_commands.js#L88)